ouroboros-network-framework/testlib/Ouroboros/Network/ConnectionManager/Test/Utils.hs

```diff
@@ -284,6 +284,14 @@ abstractStateIsFinalTransition :: Transition' AbstractState -> Bool
 abstractStateIsFinalTransition (Transition _ UnknownConnectionSt) = True
 abstractStateIsFinalTransition _                                  = False
 
+-- | This function is necessary to check tests that only read from the
+-- `traceTVar` logging output. These `traceTVar` logs do not transition to
+-- 'UnknownConnectionSt'.
```

Comment:
```suggestion
-- | This function is necessary to check tests that only read from the
-- `traceTVar` logging output. These `traceTVar` logs do not transition to
-- 'UnknownConnectionSt', since `UnknowConnectionSt` reflects the removal
-- of the `TVar` from the state map.
```

Reply:
It's an interesting limitation of `traceTVar`.

We could trace the `any -> UnknownConnectionSt` transition using `traceM`.  Things cannot be reordered since this is always the last transition, so it should work in `IOSimPOR`.  The easiest way to do it would be to pass a `traceM` tracer which ignores all other transitions (so we don't get duplicate events).